### ISSUE TITLE
Flask app can't start another flask app because of werkzeug issue

### ISSUE THREAD

**Issue by timrprobocom** [pallets/werkzeug#2067]
If a flask app tries to start another flask app using the Python `subprocess` module, the call will fail.  Werkzeug in the first Flask process has saved its file descriptor in the environment variable `WERKZEUG_SERVER_FD`.  When the second process initializes, it inherits that environment variable, but not the fd, eventually resulting in a "bad file descriptor" error when werkzeug tries to dup it.

As a workaround, I instructed the user to `del` the variable from os.environ before starting the flask app.

I don't understand the purpose of the environment variable; you may decide it is more important than this relatively obscure edge case.

StackOverflow link to the original problem:
https://stackoverflow.com/questions/66665174/run-a-server-from-a-flask-route/

```
Traceback (most recent call last):
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/pydevd.py", line 1758, in <module>
    main()
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/pydevd.py", line 1752, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/pydevd.py", line 1147, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/Applications/PyCharm CE.app/Contents/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/galshahar/Desktop/Projects/PychramProjects/cba/src/api.py", line 48, in <module>
    app.run(port=8080, debug=True)
  File "/Users/galshahar/anaconda3/lib/python3.7/site-packages/flask/app.py", line 943, in run
    run_simple(host, port, self, **options)
  File "/Users/galshahar/anaconda3/lib/python3.7/site-packages/werkzeug/serving.py", line 814, in run_simple
    inner()
  File "/Users/galshahar/anaconda3/lib/python3.7/site-packages/werkzeug/serving.py", line 774, in inner
    fd=fd)
  File "/Users/galshahar/anaconda3/lib/python3.7/site-packages/werkzeug/serving.py", line 660, in make_server
    passthrough_errors, ssl_context, fd=fd)
  File "/Users/galshahar/anaconda3/lib/python3.7/site-packages/werkzeug/serving.py", line 574, in __init__
    socket.SOCK_STREAM)
  File "/Users/galshahar/anaconda3/lib/python3.7/socket.py", line 463, in fromfd
    nfd = dup(fd)
OSError: [Errno 9] Bad file descriptor
```

**Comment by davidism** [comment]
This is absolutely not a bug.  I encourage you to reconsider posting an answer that in any way enables them to do what they're trying to do, it is a very bad idea. It's not clear what problem they're actually trying to solve with it, but this is definitely not the answer.

**Comment by davidism** [comment]
Inheriting the file descriptor should be fixed in Werkzeug 2.1. The linked answer is even less of a good idea now, it will break the reloader.
